CourseGrab/Home/HomeTableViewController.swift

```diff
@@ -114,6 +66,42 @@ class HomeTableViewController: UITableViewController {
 
 }
 
+// MARK: - Networking
+
+extension HomeTableViewController {
+
+    private func getAllTrackedCourses() {
+        NetworkManager.shared.getAllTrackedCourses().observe { result in
+            switch result {
+            case .value(let response):
+                let available = response.data.filter { $0.status == .open }
+                if !available.isEmpty {
+                    self.tableSections.append(.available(available))
+                }
+                let awaiting = response.data.filter { $0.status != .open }
+                if !awaiting.isEmpty {
+                    self.tableSections.append(.awaiting(awaiting))
+                }
+                var newSections: IndexSet = []
+
+                if newSections.count > 0 {
+                    (0..<self.tableSections.count).forEach { newSections.insert($0) }
+                    self.tableView.insertSections(newSections, with: .automatic)
+                } else {
+                    self.showEmptyState()
+                }
+            case .error(let error):
+                print(error)
```

Comment:
Would this be where we'd want to present the "no internet" or error view in HomeTableViewController?

Reply:
Yes because in Zeplin it is the background view of the table, not an alert or anything like that.